feat: add default language support

GoodbyeNJN opened the pull request:
This commit adds an option to specify a default language.

Currently, if no language is specified when writing code block or inline code, the plugin will simply skip highlighting.

In the above case, if user specifies a default language, the plugin will use default language for highlighting; if user does not specify a default language, the plugin behavior will be consistent with the current one.

atomiks commented:
I've also thought of this feature but wasn't sure how useful it was in practice, which only applies when writing docs with multiple languages (e.g. JS/CSS/HTML all on the same page). 

I wondered if there was a better way to auto-detect lang, or maybe be more granular with the option by allowing you to apply the default lang conditionally based on certain things. e.g. a Theming page would prefer a CSS default lang, while an API page a JS default lang.

Before we merge, this change needs tests for both block and inline code

GoodbyeNJN commented:
Reopen PR with these changes:
1. Merge new commits in master
2. Split `defaultLang` option into two options, `defaultCodeBlockLang` and `defaultInlineCodeLang`
3. Fixed condition issue when check code blocks

atomiks commented:
@GoodbyeNJN thank you, this needs some tests

Also I think it would be better if the option was this type, like the `theme` one:

```js
defaultLang: string | { block: string?, inline?: string }
```

GoodbyeNJN commented:
> @GoodbyeNJN thank you, this needs some tests
> 
> Also I think it would be better if the option was this type, like the `theme` one:
> 
> ```js
> defaultLang: string | { block?: string; inline?: string }
> ```

Yeah, it does look better, I'll update code.
